src/fn.php

```diff
@@ -506,9 +510,13 @@ function indexOf(callable $predicate, iterable $iter) {
 function trans(callable $trans, callable $fn, $data) {
     return $fn($trans($data));
 }
-function not(callable $fn, ...$args): bool {
-    return !$fn(...$args);
+
+function complement(callable $fn): callable {
```

Comment:
Ah, so similar to our discussion around method vs invoker. I think complement here i just the curried version of not. Can you make it explicit that these are the same by making the implementation:

```
function complement(callable $fn): callable {
  return \Krak\Fun\Curried\not($fn);
}
```

Reply:
Well, I'm not a fan of the current version of `not`, so I'd rather not tie `complement` to it. A similar thing happens with `invoker` and `method`. Plus, the code is already written, so the only benefits would be in terms of maintainability, which would be marginal (if any). If you still feel those changes are for the best, I think you have permission to edit the pull request(s)

Reply:
Ya, the reality of it is that these are just aliases of functionality that currently exists. I can't just add any method without trying to make it explicit to someone reading the docs and source code why a function exists and how it relates to other functions that do the exact same thing.

Anywho, If these are going to be merged, I'll need to be made abundantly clear that these are just aliases and offer the same functionality. Otherwise, I believe it would have a negative impact on the discoverability and comprehension of the documentation.

Reply:
I feel like one function being equivalent to the curried version of another is not exactly the same as being an alias for that function. For example, you can simulate `f\not($fn, 1, 2)` with `f\complement($fn)(1, 2)` but you can't simulate `c\complement` with `not` (not a common use case but you might want to do stuff like `map(c\complement, [$fn1, $fn2])`).

This all stems from the fact that I feel like the curried versions of `not` and `method` should be the default ones. For example, with `not`, when I go looking for a function that returns the complement of another function, I don't expect its default behavior to also call that function.

Perhaps they could be linked the other way around? Like
```php
 function not(callable $fn, ...$args) : bool {
    return \Krak\Fun\complement($fn)(...$args);
}
```

Reply:
I understand your sentiment, but all of the non-curried versions are defined first so that the curried versions can be generated. I don't have curried versions that get moved to uncurried form.

Anyways, I prefer not and method as the defaults. if you're not willing to make that change, that's fine, but I likely won't merge otherwise.